storage/aptosdb/src/lib.rs

```diff
@@ -1414,6 +1414,34 @@ impl DbWriter for AptosDB {
             )
         })
     }
+
+    fn delete_genesis(&self) -> Result<()> {
+        gauged_api("delete_genesis", || {
+            // Deletion range: (begin_version, end_version]
+            let begin_version = 0;
+            let end_version = 1;
+
+            // Delete transaction schema
+            let mut db_batch = SchemaBatch::new();
+            self.transaction_store.prune_transaction_schema(
+                begin_version,
+                end_version,
+                &mut db_batch,
+            )?;
+
+            // Delete transaction info schema
+            self.transaction_store.prune_transaction_info_schema(
+                begin_version,
+                end_version,
+                &mut db_batch,
+            )?;
+
+            // Delete transaction output schema
+            self.transaction_store
+                .prune_write_set(begin_version, end_version, &mut db_batch)?;
+            self.db.write_schemas(db_batch)
+        })
```

Comment:
Probably the event store, state db, etc. as well..

Reply:
@msmouse, I started to do this, but realized it was getting really long copying and pasting all the lines from the various pruners 😄 So, I added a comment to the Trait itself, but maybe I should add it here, too? State sync only cares about txns, infos and txn outputs when deciding availability. Everything else is irrelevant. So I was just going to rely on the DB pruner to come and clean up the rest of the stuff.

Reply:
@JoshLind - Ideally, we should delete all the relevant data from event store, state store etc to keep the DB in a consistent state. Currently these pruners are working independently, so you are right that the DB pruner will take care of deleting the rest but I plan to make pruners dependent on each other - so for example the transaction pruner will guarantee that the event store is also up to date - this is to avoid expensive initialization for those pruners. Instead of copying a bunch of code - may be you can just call the `pruner.prune` function for the pruners - it might require a bit of refactoring but that seems like the cleanest approach here - I am happy to help here if needed? Thoughts?

Reply:
@JoshLind - I looked into it and it seems like it should be fairly trivial for you to create instances of the pruners here and just call the `pruner.prune` function. I would prefer if we can do that part in this PR - I can later look into a way to refactor the pruner code so that we don't need to create so many pruner instances here.